src/nnet_packet_bindings.cpp

```diff
@@ -28,24 +30,196 @@ void init_binding_nnet_packet(pybind11::module& m){
         }, py::keep_alive<0, 1>()) /* Keep list alive while iterator is used */
         ;
 
-
     // for NNET_PACKET in nnet_packets:
     py::class_<NNetPacket, std::shared_ptr<NNetPacket>>(m, "NNetPacket")
-        .def("get_tensor", static_cast<py::array* (NNetPacket::*)(unsigned)>(&PyNNetPacket::getTensor), py::return_value_policy::copy)
-        .def("get_tensor", static_cast<py::array* (NNetPacket::*)(const std::string&)>(&PyNNetPacket::getTensorByName), py::return_value_policy::copy)
-        .def("entries", &NNetPacket::getTensorEntryContainer, py::return_value_policy::copy)
+        .def("get_tensor", static_cast<py::array* (NNetPacket::*)(unsigned)>(&PyNNetPacket::getTensor), py::return_value_policy::take_ownership)
+        .def("get_tensor", static_cast<py::array* (NNetPacket::*)(const std::string&)>(&PyNNetPacket::getTensorByName), py::return_value_policy::take_ownership)
+        .def("__getitem__", static_cast<py::array* (NNetPacket::*)(unsigned)>(&PyNNetPacket::getTensor), py::return_value_policy::take_ownership)
+        .def("__getitem__", static_cast<py::array* (NNetPacket::*)(const std::string&)>(&PyNNetPacket::getTensorByName), py::return_value_policy::take_ownership)
+        // .def("entries", &NNetPacket::getTensorEntryContainer, py::return_value_policy::copy)
         .def("getMetadata", &NNetPacket::getMetadata, py::return_value_policy::copy)
+        .def("getOutputsList", static_cast<std::list<py::array*> (NNetPacket::*)()>(&PyNNetPacket::getOutputsList), py::return_value_policy::take_ownership)     
+        .def("getOutputsDict", static_cast<std::map<std::string, py::array*> (NNetPacket::*)()>(&PyNNetPacket::getOutputsDict), py::return_value_policy::take_ownership)     
+        .def("getTensorsSize", &NNetPacket::getTensorsSize, py::return_value_policy::copy)     
+        .def("getOutputLayersInfo", &NNetPacket::getOutputLayersInfo, py::return_value_policy::copy)
+        .def("getInputLayersInfo", &NNetPacket::getInputLayersInfo, py::return_value_policy::copy)
+        .def("getDetectedObjects", &NNetPacket::getDetectedObjects)  
+        ;
+    
+    py::class_<TensorDataType>(m, "TensorDataType")
+        .def("__str__", [](const TensorDataType& v) {
+            return type_to_string.at(v);
+        })
+        .def("__repr__", [](const TensorDataType& v) {
+            return type_to_string.at(v);
+        })
+        ;
+
+    py::enum_<dimension>(m, "dimension")
```

Comment:
As in depthai-shared that could then rather be `py::enum_<Dimension>(m, "Dimension")`